python/ray/tune/tuner.py

```diff
@@ -144,27 +145,48 @@ def __init__(
                 ).remote(**kwargs)
 
     @classmethod
-    def restore(cls, path: str) -> "Tuner":
+    def restore(
+        cls,
+        path: str,
+        resume_unfinished: bool = True,
+        resume_errored: bool = False,
+        restart_errored: bool = False,
+    ) -> "Tuner":
         """Restores Tuner after a previously failed run.
 
         Args:
-           path: The path where the previous failed run is checkpointed.
-               This information could be easily located near the end of the
-               console output of previous run.
-               Note: depending on whether ray client mode is used or not,
-               this path may or may not exist on your local machine.
+            path: The path where the previous failed run is checkpointed.
+                This information could be easily located near the end of the
+                console output of previous run.
+                Note: depending on whether ray client mode is used or not,
+                this path may or may not exist on your local machine.
+            resume_unfinished: If True, will continue to run unfinished trials.
+            resume_errored: If True, will re-schedule errored trials and try to
+                restore from their latest checkpoints.
+            restart_errored: If True, will re-schedule errored trials but force
+                restarting them frmo scratch (no checkpoint will be loaded).
```

Comment:
Typo in "from"

How do these three interact with each other? I think we should document this. For example, what happens if you do `resume_errored=True` and `restart_errored=True`? etc

Reply:
OK just thought this through. So basically there are 3 groups of trials:

Finished, Unfinished, and Errored.

For Finished, we should skip them. Nothing to do.
For Unfinished, we should provide the options {skip, resume}.
For Errored, we should provide {skip, restart, resume from checkpoint}.

And all 3 groups of trials should be orthogonal to each other.

Can we provide a grouping similar to the above?

Reply:
Isn't this exactly what we currently do?

- Finished are always added and show up as TERMINATED in the stats. They are not re-run but users will have them in their results table
- Unfinished: We have `resume_unfinished` for this
- Errored: We have `resume_errored` and `restart_errored` for this. If None are set, they are skipped (turn up as ERRORed in the result). If resume is set, they are resumed from checkpoint. If restart is set, they are restarted from checkpoint

Reply:
Or is the naming/documentation confusing?

Reply:
I've added a more extensive docstring to explain the behavior

Reply:
sounds good, what i meant was to have 1 flag for the errored, not two.

Reply:
This should resolve #24918

Reply:
One thing I remember:

When using the `trial_dirname_creator`  in `tune.run()`  the resumed experiment did not use it, while the errored one did. Did anyone also run into this issue?